openedx/core/djangolib/js_utils.py

```diff
@@ -69,7 +69,12 @@ def dump_js_escaped_json(obj, cls=EdxJSONEncoder):
         (string) Escaped encoded JSON.
 
     """
-    json_string = json.dumps(obj, ensure_ascii=True, cls=cls)
+    try:
+        json_string = json.dumps(obj, ensure_ascii=True, cls=cls)
+    except:  # pylint: disable=bare-except
+        # in some cases dict_values appear here.
+        ls = [v for v in obj]
```

Comment:
Can you provide more information on what's happening here?  It looks like if we can't dump the obj as ascii, you make a list of keys and only dump the keys?

Reply:
@feanil in some cases obj appears as ```dict_values```
```
 json.dumps(obj, ensure_ascii=True, cls=cls)
*** TypeError: dict_values([{'color': '#b72121', 'data': [[1, 0.0], [2, 0.0], [3, 0.0], [4, 0.0], [5, 0.0], [6, 0.0], [7, 0.0], [8, 0.0], [9, 0.0], [10, 0.0], [11, 0.0], [12, 0.0], [13.25, 0.0]], 'label': 'Homework'}, {'color': '#666666', 'data': [[28.25, 0.0]], 'label': 'Midterm Exam'}, {'color': '#600101', 'data': [[14.5, 0.0], [15.5, 0.0], [16.5, 0.0], [17.5, 0.0], [18.5, 0.0], [19.5, 0.0], [20.5, 0.0], [21.5, 0.0], [22.5, 0.0], [23.5, 0.0], [24.5, 0.0], [25.5, 0.0], [26.75, 0.0]], 'label': 'Lab'}, {'color': '#333333', 'data': [[29.75, 0.0]], 'label': 'Final Exam'}]) is not JSON serializable
```

Reply:
@feanil I made change [here](https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/progress_graph.js#L103).

and its values is passing [here](https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/progress_graph.js#L140)

Reply:
Rather than doing a list comprention you can just do `json.dumps(list(ls), ensure_ascii=True, cls=cls)`